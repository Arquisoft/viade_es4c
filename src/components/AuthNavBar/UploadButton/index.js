import {UploadComponent} from "./upload.component";

export default UploadComponent;