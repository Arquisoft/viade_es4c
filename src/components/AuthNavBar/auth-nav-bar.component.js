import React from "react";
import {Image, Navbar} from "react-bootstrap";
import Nav from "react-bootstrap/Nav";
import UploadButton from "./UploadButton";
import auth from "solid-auth-client";
import {NavLink} from "react-router-dom";

/**
* Navigation bar which contains de actions of a user logged in
*/
const AuthNavBar = () => {

  const logOut=() => {
    auth.logout();
    window.location="/viade_es4c";
  };

  return (
      <Navbar bg="light" expand="lg">
          <Navbar.Brand href="/viade_es4c/"><Image src="/viade_es4c/img/logo.svg" alt="Viade" /></Navbar.Brand>
          <UploadButton/>
          <Navbar.Toggle aria-controls="basic-navbar-nav" />
          <Navbar.Collapse id="basic-navbar-nav">
              <Nav  className="mr-auto"/>
              <Nav>
                  <NavLink exact to="/viade_es4c/notifications">
                      <Image className="logout-img" src="/viade_es4c/img/bell.svg" alt="Notifications" />
                  </NavLink>
                  <NavLink exact to="/viade_es4c/myRoutes" className="link">MyRoutes</NavLink>
                  <NavLink exact to="/viade_es4c/friendsRoutes" className="link">FriendsRoutes</NavLink>
                  <NavLink exact to="/viade_es4c/profile" className="link">Profile</NavLink>
                  <NavLink exact to="/viade_es4c/share" className="link">Share</NavLink>
                  <Image className="logout-img" src="/viade_es4c/img/logout.svg" onClick={logOut}/>
              </Nav>
          </Navbar.Collapse>
      </Navbar>
  );
};

export default AuthNavBar;
