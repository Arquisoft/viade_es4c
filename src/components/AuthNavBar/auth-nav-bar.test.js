import React from "react";
import {cleanup, render} from "react-testing-library";
import {HashRouter as Router} from "react-router-dom";
import AuthNavBar from "./auth-nav-bar.component";

describe.only("AuthNavBar", () => {
  afterAll(cleanup);

  const { container } = render(
    <Router>
      <AuthNavBar t={key => key} />
    </Router>
  );

  test("renders without crashing", () => {
    expect(container).toBeTruthy();
  });

  test("logout", () => {
    const { getByTestId } = render(<Router><AuthNavBar /></Router>);
    const inputRoute = getByTestId("logoutbtn");
    inputRoute.click();
  });

});