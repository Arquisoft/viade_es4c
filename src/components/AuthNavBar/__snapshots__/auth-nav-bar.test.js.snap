// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthNavBar should render correctly in "debug" mode 1`] = `
<Navbar
  bg="light"
  collapseOnSelect={false}
  expand="lg"
  variant="light"
>
  <NavbarBrand
    href=""
  >
    <Image
      alt="Viade"
      fluid={false}
      rounded={false}
      roundedCircle={false}
      src="/img/logo.svg"
      thumbnail={false}
    />
  </NavbarBrand>
  <UploadButton />
  <NavbarToggle
    aria-controls="basic-navbar-nav"
    label="Toggle navigation"
  />
  <NavbarCollapse
    id="basic-navbar-nav"
  >
    <Nav
      className="mr-auto"
      fill={false}
      justify={false}
    />
    <Nav
      fill={false}
      justify={false}
    >
      <NavLink
        exact={true}
        to="/notifications"
      >
        <Image
          alt="Notifications"
          className="logout-img"
          fluid={false}
          rounded={false}
          roundedCircle={false}
          src="/img/bell.svg"
          thumbnail={false}
        />
      </NavLink>
      <NavLink
        className="link"
        exact={true}
        to="/myRoutes"
      >
        MyRoutes
      </NavLink>
      <NavLink
        className="link"
        exact={true}
        to="/friendsRoutes"
      >
        FriendsRoutes
      </NavLink>
      <NavLink
        className="link"
        exact={true}
        to="/profile"
      >
        Profile
      </NavLink>
      <NavLink
        className="link"
        exact={true}
        to="/share"
      >
        Share
      </NavLink>
      <Image
        className="logout-img"
        fluid={false}
        onClick={[Function]}
        rounded={false}
        roundedCircle={false}
        src="/img/logout.svg"
        thumbnail={false}
      />
    </Nav>
  </NavbarCollapse>
</Navbar>
`;
